From 63740552ab79f57bb92147657737afc24979e78e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bastien=20ROUCARI=C3=88S?= <roucaries.bastien@gmail.com>
Date: Wed, 15 Jun 2011 23:51:24 +0200
Subject: [PATCH] Replace a memcpy by a memmove

Overlapped memory segment could trigger bug in recent gcc/glibc. Prevent it.

cherry picked from commit f547649477576f8037a8c6b63bb45a1cedf4b992

from: John Cristy
Last-Update: 2011-06-15
Applied-Upstream: 6.7.0.1
git-svn-id: https://www.imagemagick.org/subversion/ImageMagick/trunk@4559 aa41f4f7-0bf4-0310-aa73-e5a19afd5a74
---
 magick/string.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/magick/string.c b/magick/string.c
index 70322f2..24de39d 100644
--- a/magick/string.c
+++ b/magick/string.c
@@ -2329,7 +2329,7 @@ MagickExport void StripString(char *message)
   if (q > p)
     if ((*q == '\'') || (*q == '"'))
       q--;
-  (void) memcpy(message,p,(size_t) (q-p+1));
+  (void) memmove(message,p,(size_t) (q-p+1));
   message[q-p+1]='\0';
   for (p=message; *p != '\0'; p++)
     if (*p == '\n')
-- 
1.7.5.3

